### ISSUE TITLE
[Bug] missing image on email alert notification after Grafana 5 upgrade

### ISSUE THREAD

**Issue by M-M-M-M** [grafana/grafana#11054]
-  Grafana version : 5 (grafana-5.0.0-1.x86_64)
-  datasource : influxdb
- OS : CentOS Linux release 7.4.1708 (Core)
- I was on previous version of Grafana (4) and after a yum update I have upgraded to Grafana 5. I had several alert configured with notification by email.
- The expected result is to receive alert notification email as before including image of the graphs. 
- Instead I am receiving emails but without the image. 
-  on grafana.ini I have tried to add local for provider as below but no effect.
```
[external_image_storage]
# Used for uploading images to public servers so they can be included in slack/email messages.
# you can choose between (s3, webdav)
;provider =
provider = local
```

**Comment by daniellee** [comment]
I might be able to recreate this. I tested without the `[external_image_storage]` section in my config and it worked with embedded images. When I tested with your config then I got a broken link.

Logs when it works:

```
DBUG[03-04|15:19:11] [phantom] Loading a web page: http://localhost:3000/d-solo/000000371/lazy-load-influxdb-templated-queries?panelId=1 status: success 30000 logger=png-renderer
DBUG[03-04|15:19:11] Image rendered                           logger=png-renderer path=/home/daniel/dev/go/src/github.com/grafana/grafana/data/png/aCGAd7sops19CG3txVrR.png
INFO[03-04|15:19:11] uploaded                                 logger=alerting.notifier url=http://localhost:3000/public/img/attachments/aCGAd7sops19CG3txVrR.png
```

Logs with provider set to local - note that alerting.notifier url is empty:

```
DBUG[03-04|15:21:41] Image rendered                           logger=png-renderer path=/home/daniel/dev/go/src/github.com/grafana/grafana/data/png/RoZkrVb5jXbXv7KkxOCr.png
INFO[03-04|15:21:41] uploaded                                 logger=alerting.notifier url=
```
Do your logs look similar or are you seeing something else?

**Comment by M-M-M-M** [comment]
I have activated debug level for log but I am not seeing anything related to Image rendered.

I am getting this kind of lines when an alert is triggered:

```
t=2018-03-05T12:26:15+0000 lvl=info msg="Alert Rule returned no data" logger=alerting.evalHandler ruleId=7 name="Download file alert" changing state to=no_data
t=2018-03-05T12:26:15+0000 lvl=info msg="New state change" logger=alerting.resultHandler alertId=7 newState=no_data prev state=ok
t=2018-03-05T12:26:15+0000 lvl=dbug msg="Sending notification" logger=alerting.notifier type=email id=1 isDefault=true
t=2018-03-05T12:26:15+0000 lvl=info msg="Sending alert notification to" logger=alerting.notifier.email addresses=[example@iexample.net]
t=2018-03-05T12:26:15+0000 lvl=dbug msg="Job Execution completed" logger=alerting.engine timeMs=6.772 alertId=7 name="Download alert" firing=false
```

**Comment by bergquist** [comment]
Needs better docs

**Comment by bergquist** [comment]
```
[external_image_storage] 
provider = local
```
might not always work with email notifications since email server/clients might not be able to access the grafana instance. I will mention this in the docs and close this.

**Comment by bergquist** [comment]
closed by 2743e8be20cc36d48370f78df140a42ff209cb60

**Comment by M-M-M-M** [comment]
So something that was working in previous version is no more working and considered as a feature?

**Comment by M-M-M-M** [comment]
This has been resolved with the release grafana-5.0.4-1.x86_64.

**Comment by drix1994** [comment]
Hello, 

> closed by [2743e8b](https://github.com/grafana/grafana/commit/2743e8be20cc36d48370f78df140a42ff209cb60)

So could you explain how grafana sends the notifications via email? 
I launched Grafana Insance in internal network where we use internal mail server too and when I get notification from Grafana, I don't have a image or url to img file, the same is for alert rule. 
Why ?? I dont understand why Grafana need access to internet ?? Maybe I need to open some specifics ports on FW ??
